pyvista/plotting/axes_actor.py

```diff
@@ -1,176 +1,416 @@
+from pyvista.core.errors import PyVistaDeprecationWarning
+from pyvista.core.utilities.arrays import (
+    _coerce_transformlike_arg,
+    array_from_vtkmatrix,
+    vtkmatrix_from_array,
+)
+from pyvista.core.utilities.misc import AnnotatedIntEnum
+from pyvista.core.utilities.transformations import apply_transformation_to_points
+
+from ._vtk import vtkAxesActor, vtkMatrix3x3, vtkMatrix4x4, vtkTransform
 from .actor_properties import ActorProperties
+from .colors import Color, ColorLike
+from .prop3d import Prop3D
 
 
-class AxesActor(_vtk.vtkAxesActor):
-    """Axes actor wrapper for vtkAxesActor.
-
-    Hybrid 2D/3D actor used to represent 3D axes in a scene. The user
-    can define the geometry to use for the shaft or the tip, and the
-    user can set the text for the three axes. To see full customization
-    options, refer to `vtkAxesActor Details
-    <https://vtk.org/doc/nightly/html/classvtkAxesActor.html#details>`_.
+class AxesActor(Prop3D, vtkAxesActor):  # numpydoc ignore=PR01
+    """Wrapper for vtkAxesActor.
 
-    Examples
-    --------
-    Customize the axis shaft color and shape.
+    Hybrid 2D/3D actor used to represent 3D axes in a scene.
 
-    >>> import pyvista as pv
+    """
 
-    >>> axes = pv.Axes()
-    >>> axes.axes_actor.z_axis_shaft_properties.color = (0, 1, 1)
-    >>> axes.axes_actor.shaft_type = axes.axes_actor.ShaftType.CYLINDER
-    >>> pl = pv.Plotter()
-    >>> _ = pl.add_actor(axes.axes_actor)
-    >>> _ = pl.add_mesh(pv.Sphere())
-    >>> pl.show()
+    class ShaftType(AnnotatedIntEnum):
+        """Types of shaft shapes available."""
 
-    Or you can use this as a custom orientation widget with
-    :func:`add_orientation_widget() <pyvista.Renderer.add_orientation_widget>`:
+        CYLINDER = (0, "cylinder")
+        LINE = (1, "line")
 
-    >>> import pyvista as pv
+    class TipType(AnnotatedIntEnum):
+        """Types of tip shapes available."""
 
-    >>> axes = pv.Axes()
-    >>> axes_actor = axes.axes_actor
-    >>> axes.axes_actor.shaft_type = 0
+        CONE = (0, "cone")
+        SPHERE = (1, "sphere")
+
+    def __init__(
+        self,
+        x_label='X',
+        y_label='Y',
+        z_label='Z',
+        label_color=None,
+        labels_off=False,
+        label_size=(0.25, 0.1),
+        label_position=1,
+        x_color=None,
+        y_color=None,
+        z_color=None,
+        shaft_type="cylinder",
+        shaft_length=0.8,
+        shaft_radius=0.01,
+        shaft_width=2,
+        shaft_resolution=16,
+        tip_radius=0.4,
+        tip_length=0.2,
+        tip_type="cone",
+        tip_resolution=16,
+        total_length=(1, 1, 1),
+        position=(0, 0, 0),
+        orientation=(0, 0, 0),
+        origin=(0, 0, 0),
+        scale=(1, 1, 1),
+        user_matrix=np.eye(4),
+        visibility=True,
+        properties=None,
+        **kwargs,
+    ):
+        """Create a hybrid 2D/3D actor to represent 3D axes in a scene.
```

Comment:
I think we want the docstrings on the class, not the `__init__`. At least this is how it is done everywhere else in the docs for PyVista. I believe as it stands, your docstring will not be rendered in our documentation.

Note: this is confusing and I recently opened an issue around the topic: #5137